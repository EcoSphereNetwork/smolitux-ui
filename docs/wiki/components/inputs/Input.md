# Input Komponente

Die Input-Komponente ist ein grundlegendes Formular-Element für die Texteingabe mit umfangreichen Anpassungsmöglichkeiten und Barrierefreiheitsfunktionen.

## Eigenschaften

| Eigenschaft          | Typ                                                | Standard    | Beschreibung                                                |
| -------------------- | -------------------------------------------------- | ----------- | ----------------------------------------------------------- |
| `label`              | `ReactNode`                                        | -           | Text-Label für das Input-Feld                               |
| `helperText`         | `ReactNode`                                        | -           | Hilfetext unter dem Input-Feld                              |
| `error`              | `ReactNode`                                        | -           | Fehlermeldung                                               |
| `successMessage`     | `ReactNode`                                        | -           | Erfolgsmeldung                                              |
| `leftIcon`           | `ReactNode`                                        | -           | Icon links im Input-Feld                                    |
| `rightIcon`          | `ReactNode`                                        | -           | Icon rechts im Input-Feld                                   |
| `size`               | `'xs' \| 'sm' \| 'md' \| 'lg' \| 'xl'`             | `'md'`      | Größe des Input-Felds                                       |
| `variant`            | `'outline' \| 'filled' \| 'flushed' \| 'unstyled'` | `'outline'` | Visuelle Variante                                           |
| `type`               | `InputType`                                        | `'text'`    | HTML-Input-Typ (text, password, email, etc.)                |
| `fullWidth`          | `boolean`                                          | `false`     | Input nimmt die volle verfügbare Breite ein                 |
| `isLoading`          | `boolean`                                          | `false`     | Zeigt einen Ladezustand an                                  |
| `isValid`            | `boolean`                                          | `false`     | Zeigt einen gültigen Zustand an                             |
| `isInvalid`          | `boolean`                                          | `false`     | Zeigt einen ungültigen Zustand an                           |
| `isSuccess`          | `boolean`                                          | `false`     | Zeigt einen erfolgreichen Zustand an                        |
| `isDisabled`         | `boolean`                                          | `false`     | Deaktiviert das Input-Feld                                  |
| `isRequired`         | `boolean`                                          | `false`     | Markiert das Input-Feld als erforderlich                    |
| `isReadOnly`         | `boolean`                                          | `false`     | Macht das Input-Feld schreibgeschützt                       |
| `showPasswordToggle` | `boolean`                                          | `false`     | Zeigt einen Button zum Umschalten der Passwort-Sichtbarkeit |
| `isClearable`        | `boolean`                                          | `false`     | Zeigt einen Button zum Löschen des Inhalts                  |
| `prefix`             | `ReactNode`                                        | -           | Präfix vor dem Text (z.B. Währungssymbol)                   |
| `suffix`             | `ReactNode`                                        | -           | Suffix nach dem Text (z.B. Einheit)                         |
| `leftAddon`          | `ReactNode`                                        | -           | Element links neben dem Eingabefeld                         |
| `rightAddon`         | `ReactNode`                                        | -           | Element rechts neben dem Eingabefeld                        |
| `showCounter`        | `boolean`                                          | `false`     | Zeigt einen Zeichenzähler an                                |
| `maxLength`          | `number`                                           | -           | Maximale Anzahl von Zeichen                                 |
| `showProgressBar`    | `boolean`                                          | `false`     | Zeigt einen Fortschrittsbalken an                           |
| `onChange`           | `(e: React.ChangeEvent<HTMLInputElement>) => void` | -           | Callback bei Wertänderung                                   |
| `onFocus`            | `(e: React.FocusEvent<HTMLInputElement>) => void`  | -           | Callback bei Fokussierung                                   |
| `onBlur`             | `(e: React.FocusEvent<HTMLInputElement>) => void`  | -           | Callback bei Fokusverlust                                   |

## Beispiele

### Grundlegende Verwendung

```jsx
<Input
  label="Email"
  placeholder="name@example.com"
  type="email"
  helperText="Wir werden Ihre Email niemals teilen."
/>
```

### Mit Icons

```jsx
<Input
  label="Suche"
  placeholder="Suchbegriff eingeben"
  leftIcon={<SearchIcon />}
  rightIcon={<ClearIcon />}
  isRightIconClickable
  onRightIconClick={() => setValue('')}
/>
```

### Passwort-Feld mit Toggle

```jsx
<Input label="Passwort" type="password" showPasswordToggle showCounter maxLength={20} />
```

### Validierungszustände

```jsx
<Input
  label="Benutzername"
  isValid
  successMessage="Benutzername ist verfügbar"
/>

<Input
  label="Email"
  type="email"
  isInvalid
  error="Bitte geben Sie eine gültige Email-Adresse ein"
/>
```

### Mit Präfix und Suffix

```jsx
<Input
  label="Preis"
  type="number"
  prefix="€"
  placeholder="0.00"
/>

<Input
  label="Gewicht"
  type="number"
  suffix="kg"
  placeholder="0"
/>
```

### Mit Addons

```jsx
<Input placeholder="Website" leftAddon="https://" />
<Input placeholder="Benutzername" rightAddon="@example.com" />
<Input placeholder="Preis" leftAddon="€" rightAddon=".00" />
```

### Größenvarianten

```jsx
<Input size="xs" placeholder="Extra klein" />
<Input size="sm" placeholder="Klein" />
<Input size="md" placeholder="Mittel (Standard)" />
<Input size="lg" placeholder="Groß" />
<Input size="xl" placeholder="Extra groß" />
```

### Stilistische Varianten

```jsx
<Input variant="outline" placeholder="Outline (Standard)" />
<Input variant="filled" placeholder="Filled" />
<Input variant="flushed" placeholder="Flushed" />
<Input variant="unstyled" placeholder="Unstyled" />
```

## Barrierefreiheit

Die Input-Komponente wurde mit besonderem Fokus auf Barrierefreiheit entwickelt:

- Korrekte Verknüpfung von Labels mit Input-Feldern über `id` und `htmlFor`
- Unterstützung für `aria-describedby` für Hilfetexte und Fehlermeldungen
- Korrekte `aria-invalid` und `aria-valid` Attribute für Validierungszustände
- `aria-required` für erforderliche Felder
- `aria-disabled` für deaktivierte Felder
- `aria-readonly` für schreibgeschützte Felder
- `aria-busy` für Ladezustände
- Screenreader-freundliche Fehlermeldungen mit `aria-errormessage`
- Tastaturnavigation und -bedienung

## Design-Überlegungen

- Labels sollten klar und präzise sein
- Hilfetexte sollten zusätzliche Informationen bieten, ohne zu überwältigen
- Fehlermeldungen sollten spezifisch und lösungsorientiert sein
- Icons sollten die Funktion des Inputs unterstützen, nicht ablenken
- Validierungszustände sollten klar erkennbar sein
- Konsistente Größen und Abstände für bessere Benutzererfahrung

## Implementierungsdetails

Die Input-Komponente verwendet intern:

- Flexbox für die Ausrichtung von Text und Icons
- CSS-Transitions für Hover- und Fokus-Effekte
- React.forwardRef für Ref-Weiterleitung
- Tailwind CSS für Styling
- FormControl-Kontext für Formularintegration
- Zustandsmanagement für Passwort-Toggle und Fokus

## Aktuelle Verbesserungen

- Verbesserte ARIA-Attribute für bessere Screenreader-Unterstützung
- Hinzufügung von aria-valid, aria-disabled, aria-required, aria-readonly und aria-busy
- Korrektur der FormControl-Kontextnutzung mit optionalem Chaining
- Verbesserte Typensicherheit

## Zukünftige Verbesserungen

- Unterstützung für Masken und Formatierung
- Erweiterte Validierungsmöglichkeiten
- Autogrow-Funktionalität für Textarea-ähnliches Verhalten
- Verbesserte Mobilgeräte-Unterstützung
- Theming-Unterstützung für benutzerdefinierte Farbpaletten
