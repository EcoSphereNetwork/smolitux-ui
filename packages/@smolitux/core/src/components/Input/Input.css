.smolitux-input-wrapper {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.smolitux-input-label {
  font-size: 0.875rem;
  font-weight: 500;
  color: var(--input-label-color, #374151);
  margin-bottom: 0.25rem;
}

.smolitux-input-container {
  position: relative;
  display: flex;
  align-items: center;
  width: 100%;
}

.smolitux-input {
  width: 100%;
  font-size: 1rem;
  line-height: 1.5;
  background-color: var(--input-bg, white);
  border: 1px solid var(--input-border-color, #d1d5db);
  border-radius: 0.375rem;
  padding: 0.5rem 0.75rem;
  transition: all 0.2s ease-in-out;
}

.smolitux-input:focus {
  outline: none;
  border-color: var(--primary-500, #3b82f6);
  box-shadow: 0 0 0 3px rgba(59, 130, 246, 0.2);
}

.smolitux-input:disabled,
.smolitux-input[readonly] {
  background-color: var(--input-disabled-bg, #f3f4f6);
  cursor: not-allowed;
  opacity: 0.7;
}

.smolitux-input-left-icon,
.smolitux-input-right-icon {
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  color: var(--input-icon-color, #6b7280);
  pointer-events: none;
}

.smolitux-input-left-icon {
  left: 0.75rem;
}

.smolitux-input-right-icon {
  right: 0.75rem;
}

.smolitux-input--with-left-icon {
  padding-left: 2.5rem;
}

.smolitux-input--with-right-icon {
  padding-right: 2.5rem;
}

.smolitux-input-helper-text {
  font-size: 0.75rem;
  color: var(--input-helper-text-color, #6b7280);
  margin-top: 0.25rem;
}

.smolitux-input-error {
  font-size: 0.75rem;
  color: var(--input-error-color, #ef4444);
  margin-top: 0.25rem;
}

.smolitux-input-success {
  font-size: 0.75rem;
  color: var(--input-success-color, #10b981);
  margin-top: 0.25rem;
}

.smolitux-input-counter {
  font-size: 0.75rem;
  color: var(--input-counter-color, #6b7280);
  margin-top: 0.25rem;
  text-align: right;
}

/* Varianten */
.smolitux-input--outline {
  background-color: transparent;
}

.smolitux-input--filled {
  background-color: var(--input-filled-bg, #f3f4f6);
  border-color: transparent;
}

.smolitux-input--filled:focus {
  background-color: var(--input-filled-focus-bg, #e5e7eb);
}

.smolitux-input--flushed {
  border-width: 0;
  border-bottom-width: 1px;
  border-radius: 0;
  padding-left: 0;
  padding-right: 0;
}

.smolitux-input--unstyled {
  border: none;
  background: transparent;
  padding: 0;
}

/* Größen */
.smolitux-input--xs {
  font-size: 0.75rem;
  padding: 0.25rem 0.5rem;
}

.smolitux-input--sm {
  font-size: 0.875rem;
  padding: 0.375rem 0.625rem;
}

.smolitux-input--md {
  font-size: 1rem;
  padding: 0.5rem 0.75rem;
}

.smolitux-input--lg {
  font-size: 1.125rem;
  padding: 0.625rem 0.875rem;
}

.smolitux-input--xl {
  font-size: 1.25rem;
  padding: 0.75rem 1rem;
}

/* Zustände */
.smolitux-input--invalid {
  border-color: var(--input-error-border-color, #ef4444);
}

.smolitux-input--invalid:focus {
  box-shadow: 0 0 0 3px rgba(239, 68, 68, 0.2);
}

.smolitux-input--valid {
  border-color: var(--input-success-border-color, #10b981);
}

.smolitux-input--valid:focus {
  box-shadow: 0 0 0 3px rgba(16, 185, 129, 0.2);
}

.smolitux-input--shadow {
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.05);
}

.smolitux-input--full-width {
  width: 100%;
}

.smolitux-input--rounded {
  border-radius: 9999px;
}

.smolitux-input--transparent {
  background-color: transparent;
}

.smolitux-input--loading {
  opacity: 0.7;
}

/* Progress Bar */
.smolitux-input-progress {
  width: 100%;
  height: 4px;
  background-color: var(--input-progress-bg, #e5e7eb);
  border-radius: 2px;
  margin-top: 0.25rem;
  overflow: hidden;
}

.smolitux-input-progress-bar {
  height: 100%;
  background-color: var(--primary-500, #3b82f6);
  transition: width 0.2s ease-in-out;
}

/* Dark Mode */
.dark .smolitux-input-label {
  color: var(--input-label-color-dark, #d1d5db);
}

.dark .smolitux-input {
  background-color: var(--input-bg-dark, #1f2937);
  border-color: var(--input-border-color-dark, #4b5563);
  color: var(--input-text-color-dark, #f9fafb);
}

.dark .smolitux-input:disabled,
.dark .smolitux-input[readonly] {
  background-color: var(--input-disabled-bg-dark, #374151);
}

.dark .smolitux-input-helper-text {
  color: var(--input-helper-text-color-dark, #9ca3af);
}

.dark .smolitux-input-counter {
  color: var(--input-counter-color-dark, #9ca3af);
}

.dark .smolitux-input--filled {
  background-color: var(--input-filled-bg-dark, #374151);
}

.dark .smolitux-input--filled:focus {
  background-color: var(--input-filled-focus-bg-dark, #4b5563);
}

.dark .smolitux-input-progress {
  background-color: var(--input-progress-bg-dark, #4b5563);
}