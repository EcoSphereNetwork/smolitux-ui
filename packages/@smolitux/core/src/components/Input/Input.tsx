import React, { forwardRef } from 'react';
import { cn } from '@smolitux/utils';

export interface InputProps {
  /** Current value */
  value: string;
  /** Change handler returning the new value */
  onChange: (value: string) => void;
  /** Input type */
  type?: string;
  /** Element id */
  id?: string;
  /** Name attribute */
  name?: string;
  /** Disabled state */
  disabled?: boolean;
  /** Placeholder text */
  placeholder?: string;
  /** Required flag */
  required?: boolean;
  /** Readonly flag */
  readOnly?: boolean;
  /** Autocomplete hint */
  autoComplete?: string;
  /** Autofocus on mount */
  autoFocus?: boolean;
  /** Accessible label */
  ariaLabel?: string;
  /** aria-describedby reference */
  ariaDescribedby?: string;
  /** Additional className */
  className?: string;
}

export const Input = forwardRef<HTMLInputElement, InputProps>(
  (
    {
      value,
      onChange,
      type = 'text',
      id,
      name,
      disabled = false,
      placeholder,
      required = false,
      readOnly = false,
      autoComplete,
      autoFocus = false,
      ariaLabel,
      ariaDescribedby,
      className,
      ...rest
    },
    ref,
  ) => (
    <input
      ref={ref}
      id={id}
      name={name}
      type={type}
      disabled={disabled}
      placeholder={placeholder}
      required={required}
      readOnly={readOnly}
      autoComplete={autoComplete}
      autoFocus={autoFocus}
      value={value}
      onChange={(e) => onChange(e.target.value)}
      aria-label={ariaLabel}
      aria-describedby={ariaDescribedby}
      className={cn(
        'smx-input border rounded px-3 py-2 text-gray-900 placeholder-gray-500',
        'focus:outline-none focus:ring-2 focus:ring-primary-500',
        disabled && 'opacity-50 cursor-not-allowed',
        className,
      )}
      {...rest}
    />
  ),
);

Input.displayName = 'Input';

export default Input;
