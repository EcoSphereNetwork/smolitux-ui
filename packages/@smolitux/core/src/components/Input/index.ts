export { Input } from './Input';
export type { InputProps, InputSize, InputVariant, InputType } from './Input';

// Fur Abwartskompatibilitat mit dem bestehenden Export
export { Input as default } from './Input';