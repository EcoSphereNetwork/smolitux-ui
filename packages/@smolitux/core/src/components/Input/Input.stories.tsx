import type { Meta, StoryObj } from '@storybook/react';
import { Input } from './Input';

const meta: Meta<typeof Input> = {
  title: 'Components/Core/Input',
  component: Input,
  args: {
    value: '',
    onChange: () => {},
  },
  argTypes: {
    onChange: { action: 'changed' },
  },
  tags: ['autodocs'],
};
export default meta;

type Story = StoryObj<typeof Input>;

export const Default: Story = {
  args: {
    placeholder: 'Type here',
  },
};

export const Password: Story = {
  args: {
    type: 'password',
    placeholder: 'Password',
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
    placeholder: 'Disabled input',
  },
};
