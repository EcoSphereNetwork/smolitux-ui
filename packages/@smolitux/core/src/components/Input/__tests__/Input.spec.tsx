import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from '../Input';

// Mock für ThemeProvider
jest.mock('@smolitux/theme', () => ({
  useTheme: jest.fn(() => ({ themeMode: 'light' })),
}));

// Mock für den FormControl-Context
jest.mock('../../FormControl', () => ({
  useFormControl: () => ({
    disabled: false,
    required: false,
    hasError: false,
    id: undefined,
    label: undefined,
    name: undefined,
    size: 'md',
    readOnly: false,
    isFocused: false,
    isValid: false,
    isInvalid: false,
    isSuccess: false,
    isLoading: false
  })
}));

describe('Input Snapshots', () => {
  // Basis-Tests
  test('renders correctly with default props', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with value', () => {
    const tree = renderer.create(<Input value="Test Value" readOnly />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with placeholder', () => {
    const tree = renderer.create(<Input placeholder="Enter text here" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Label und Hilfetexte
  test('renders correctly with label', () => {
    const tree = renderer.create(<Input label="Username" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with helper text', () => {
    const tree = renderer.create(<Input helperText="Enter your username" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with error message', () => {
    const tree = renderer.create(<Input error="Username is required" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with success message', () => {
    const tree = renderer.create(<Input successMessage="Username is available" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with hidden label', () => {
    const tree = renderer.create(<Input label="Hidden Label" hideLabel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with description for screen readers', () => {
    const tree = renderer.create(<Input description="Description for screen readers" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Zustände
  test('renders correctly when disabled', () => {
    const tree = renderer.create(<Input disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly when required', () => {
    const tree = renderer.create(<Input required />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly when readonly', () => {
    const tree = renderer.create(<Input readOnly />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with loading state', () => {
    const tree = renderer.create(<Input isLoading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with valid state', () => {
    const tree = renderer.create(<Input isValid />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with invalid state', () => {
    const tree = renderer.create(<Input isInvalid />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with success state', () => {
    const tree = renderer.create(<Input isSuccess />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Icons
  test('renders correctly with left icon', () => {
    const leftIcon = <span data-testid="left-icon">@</span>;
    const tree = renderer.create(<Input leftIcon={leftIcon} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with right icon', () => {
    const rightIcon = <span data-testid="right-icon">✓</span>;
    const tree = renderer.create(<Input rightIcon={rightIcon} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with both icons', () => {
    const leftIcon = <span data-testid="left-icon">@</span>;
    const rightIcon = <span data-testid="right-icon">✓</span>;
    const tree = renderer.create(<Input leftIcon={leftIcon} rightIcon={rightIcon} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with clickable left icon', () => {
    const leftIcon = <span data-testid="left-icon">@</span>;
    const tree = renderer.create(<Input leftIcon={leftIcon} isLeftIconClickable />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with clickable right icon', () => {
    const rightIcon = <span data-testid="right-icon">✓</span>;
    const tree = renderer.create(<Input rightIcon={rightIcon} isRightIconClickable />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with prefix', () => {
    const prefix = <span data-testid="prefix">$</span>;
    const tree = renderer.create(<Input prefix={prefix} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with suffix', () => {
    const suffix = <span data-testid="suffix">kg</span>;
    const tree = renderer.create(<Input suffix={suffix} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Größen
  test('renders correctly with extra small size', () => {
    const tree = renderer.create(<Input size="xs" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with small size', () => {
    const tree = renderer.create(<Input size="sm" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with medium size', () => {
    const tree = renderer.create(<Input size="md" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with large size', () => {
    const tree = renderer.create(<Input size="lg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with extra large size', () => {
    const tree = renderer.create(<Input size="xl" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Varianten
  test('renders correctly with outline variant', () => {
    const tree = renderer.create(<Input variant="outline" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with filled variant', () => {
    const tree = renderer.create(<Input variant="filled" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with flushed variant', () => {
    const tree = renderer.create(<Input variant="flushed" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with unstyled variant', () => {
    const tree = renderer.create(<Input variant="unstyled" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Eigenschaften
  test('renders correctly with full width', () => {
    const tree = renderer.create(<Input fullWidth />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with shadow', () => {
    const tree = renderer.create(<Input shadow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly without border', () => {
    const tree = renderer.create(<Input bordered={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly without rounded corners', () => {
    const tree = renderer.create(<Input rounded={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with transparent background', () => {
    const tree = renderer.create(<Input transparent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with counter', () => {
    const tree = renderer.create(<Input showCounter maxLength={10} value="Hello" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with progress bar', () => {
    const tree = renderer.create(<Input showProgressBar progressValue={50} progressMax={100} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with password toggle', () => {
    const tree = renderer.create(<Input type="password" showPasswordToggle />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with clearable button', () => {
    const tree = renderer.create(<Input value="Test" isClearable />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with datalist', () => {
    const tree = renderer.create(<Input list="options" datalist={['Option 1', 'Option 2', 'Option 3']} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom class name', () => {
    const tree = renderer.create(<Input className="custom-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with custom container class name', () => {
    const tree = renderer.create(<Input containerClassName="container-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with custom input container class name', () => {
    const tree = renderer.create(<Input inputContainerClassName="input-container-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with custom label class name', () => {
    const tree = renderer.create(<Input label="Label" labelClassName="label-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with custom helper text class name', () => {
    const tree = renderer.create(<Input helperText="Helper" helperTextClassName="helper-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with custom error class name', () => {
    const tree = renderer.create(<Input error="Error" errorClassName="error-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with custom success class name', () => {
    const tree = renderer.create(<Input successMessage="Success" successClassName="success-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Input-Typen
  test('renders correctly with different input types', () => {
    const types = [
      'text', 
      'password', 
      'email', 
      'number', 
      'tel', 
      'url', 
      'search', 
      'date', 
      'time', 
      'datetime-local', 
      'month', 
      'week', 
      'color'
    ];
    
    types.forEach(type => {
      const tree = renderer.create(<Input type={type} />).toJSON();
      expect(tree).toMatchSnapshot(`Input with type ${type}`);
    });
  });

  // Attribute
  test('renders correctly with id', () => {
    const tree = renderer.create(<Input id="username-input" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with name', () => {
    const tree = renderer.create(<Input name="username" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with min and max', () => {
    const tree = renderer.create(<Input type="number" min={0} max={100} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with step', () => {
    const tree = renderer.create(<Input type="number" step={0.5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with pattern', () => {
    const tree = renderer.create(<Input pattern="[A-Za-z]{3}" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with autocomplete', () => {
    const tree = renderer.create(<Input autocomplete="email" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with spellCheck', () => {
    const tree = renderer.create(<Input spellCheck={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with autoCapitalize', () => {
    const tree = renderer.create(<Input autoCapitalize="none" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with autoCorrect', () => {
    const tree = renderer.create(<Input autoCorrect="off" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with inputMode', () => {
    const tree = renderer.create(<Input inputMode="numeric" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  // Kombinationen
  test('renders correctly with multiple props combined', () => {
    const leftIcon = <span data-testid="left-icon">@</span>;
    const tree = renderer.create(
      <Input 
        label="Email"
        type="email"
        placeholder="name@example.com"
        helperText="We'll never share your email"
        required
        leftIcon={leftIcon}
        size="md"
        variant="outline"
        fullWidth
        shadow
        rounded
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('renders correctly with validation props combined', () => {
    const tree = renderer.create(
      <Input 
        label="Password"
        type="password"
        showPasswordToggle
        showCounter
        maxLength={20}
        value="password123"
        isValid
        successMessage="Strong password"
        showValidationIndicator
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});