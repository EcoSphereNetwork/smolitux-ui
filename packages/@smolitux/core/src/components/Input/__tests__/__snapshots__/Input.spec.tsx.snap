// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<input
  autoFocus={false}
  className="smx-input border rounded px-3 py-2 text-gray-900 placeholder-gray-500 focus:outline-none focus:ring-2 focus:ring-primary-500"
  disabled={false}
  onChange={[Function]}
  placeholder="x"
  readOnly={false}
  required={false}
  type="text"
  value=""
/>
`;
